dips/dip-158.md

```diff
@@ -0,0 +1,805 @@
+* "Receiver", "Receiving VASP" or "Merchant VASP" - The VASP representing the merchant which receives the funds (transaction)
+* "Customer" - The person who wishes to pay the merchant for goods or services
+* "Reference ID", "Common Payment Identifier", "Payment Identifier" - A unique identifier of the payment. It must be a UUID according to RFC4122 with "-"'s included. See also [Reference ID in DIP-1](https://github.com/diem/dip/blob/main/dips/dip-1.mdx#terminology)
+---
+# Prerequisite: Sharing Payment Identifier and Receiver Address
+A prerequisite for the off-chain negotiations is that both parties (sending and receiving VASPs) have shared a common payment identifier and the receiver address. This common payment idenbtifier is required to support a successive off-chain protocol between the VASPs. 
+This DIP specifies a couple of examples for exchanging identifiers between VASPs, however, there could be others. The protocol is agnostic to the method of exchanging the common payment identifier. 
+
+See [Appendix A](#appendix-a---prerequisite-sharing-common-payment-identifier-and-address)
+
+---
+# P2M Off-Chain Protocol Specification
+This section specifies P2M payment types, required structures, payloads and expected sequence of calls based on this protocol.
+
+Content:
+* [High Level Flow and Payment Types](#high-level-flow-and-payment-types)
+* [Sender / Receiver Interface](#sender--receiver-interface)
+* [Request / Response Payload](#request--response-payload)
+* [Payment Sequences](#payment-sequences)
+* [New Structures](#new-structures)
+* [Appendix A: Sharing Common Payment Identifier and Address](#appendix-a---prerequisite-sharing-common-payment-identifier-and-address)
+---
+## High-Level Flow and Payment Types
+---
+The high-level flow of an eCommerce P2M payment (regardless of the type) is:
+1. On the merchant's checkout page the customer chooses to pay with Diem and selects the preferred Diem wallet to use
+2. The merchant's VASP (the receiver) generates a payment request with all details needed to identify this payment later in the process and makes it available to the customer's Diem wallet (the customer's VASP or sender)
+3. The customer's VASP parses the payment request (e.g. by scanning the payment request QR) and uses the provided payment identifier to get the payment details. This is the first step of the off-chain protocol
+4. The wallet then presents the requested payment details to the customer so the customer can review the payment
+5. The customer approves the payment request in the wallet
+6. The wallet initializes the payment protocol (off-chain) which will end (if successful) with the funds being authorized or captured (depending on the payment type). Successful authorization or capture implies that the receiving VASP was satisfied with the payer information provided and the sender VASP was satisfied with the merchant information provided  
```

Comment:
```suggestion
6. The wallet initializes the payment protocol (off-chain) which will end (if successful) with the funds being authorized or charged (depending on the payment type). Successful authorization or capture implies that the receiving VASP was satisfied with the payer information provided and the sender VASP was satisfied with the merchant information provided  
```

Reply:
Updated and changed some additional phrasing